### PR TITLE
gh-109151: Enable readline in the sqlite3 CLI

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
* Issue: gh-109151

### CODE REVIEW

**Comment by erlend-aasland** [comment]
Thanks, Serhiy, this is very nice! Ideally, we should have this for 3.12. What do you think? I'm not sure it can be considered a defect/bug.

**Comment by serhiy-storchaka** [comment]
It is clearly not a bug fix. But it is so much must have feature, that it should go in 3.12.1. Let wait on the release of 3.12.0.

@Yhg1s

**Review by erlend-aasland — APPROVED** [review]

**Comment by vstinner** [comment]
This is a new feature, IMO it's too late for Python 3.12. I dislike when Python behavior changes in 3.x.y bug releases, sorry.

If you want readline in Python 3.12, you *can* write your CLI code:

```py
#import readline
import sqlite3.__main__
import sys
sqlite3.__main__.main(sys.argv[1:])
```

File: Misc/NEWS.d/next/Library/2023-09-08-19-44-01.gh-issue-109151.GkzkQu.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Enable ``readline`` editing feasibilities in the :mod:`sqlite3` CLI.
```
Comment (vstinner) [inline Misc/NEWS.d/next/Library/2023-09-08-19-44-01.gh-issue-109151.GkzkQu.rst]: English is not my first language, but "editing feasibilities" sounds surprising to me:

```suggestion
Enable ``readline`` editing features in the :ref:`sqlite3 command-line interface <sqlite3-cli>` (``python -m sqlite3``).
```

**Review by vstinner — APPROVED** [review]
LGTM, just some remarks on the NEWS entry.

**Comment by vstinner** [comment]
I forgot or didn't know that sqlite3 has a command-line interface. I created issue #109435 to document all stdlib modules which provides a command-line interface (CLI).

**Comment by serhiy-storchaka** [comment]
The sqlite3 CLI is new in 3.12. This is why I think it is a good idea to backport this change.

I always use utilites that read from the stdin with an external wrapper (`rlwrap`), but the sqlite3 CLI differs from the dis CLI or others which also can read from the stdin, -- it is intended as a REPL.

**Comment by vstinner** [comment]
> The sqlite3 CLI is new in 3.12. This is why I think it is a good idea to backport this change.

Aha, you can negociate with @Yhg1s. First land it in the main branch :-)

**Review by erlend-aasland — APPROVED** [review]

**Review by vstinner — APPROVED** [review]
LGTM.

**Comment by erlend-aasland** [comment]
Thanks, Serhiy!

**Pull request merged**